Add RFCs docs (#3367)
nf-core/website: sites/docs/src/content/docs/contributing/project_proposals.md

---
title: "Project proposals / RFCs"
subtitle: Process for proposing major new nf-core projects
---

# Summary

1. Got a big idea? Post it on [`#rfc-suggestions`](https://nfcore.slack.com/archives/C08TXM0GGMT) for quick feedback.

> 2. Create an issue on nf-core/proposals using the RFC template, explaining **_what_** you want to do.

Reviewer: ```suggestion
2. Create an **issue** on [nf-core/proposals](https://github.com/nf-core/proposals) using the RFC template, explaining **_what_** you want to do.
```

> 3. Once approved by the core team, open a PR with a full RFC text as markdown, extending the issue with details on **_how_** it should be implemented.

Reviewer: ```suggestion
3. Once approved by the core team, open a **PR** on [nf-core/proposals](https://github.com/nf-core/proposals) with a full RFC text as markdown, extending the issue with details on **_how_** it should be implemented.
```

4. Start development and update the RFC PR as needed as you go along.
5. Once the new feature ships, the RFC PR is merged and the markdown used for future reference.

# Preamble

The nf-core ‘Request for Comment’ (RFC) process is designed to give the community a voice and visibility to large projects that affect the entire community.

## Glossary

<dl>
  <dt>RFC (Request for Comment):</dt>
  <dd>A formal proposal submitted for discussion, typically involving significant changes or new features. An RFC outlines the motivation, requirements, and steps necessary for implementation, and invites feedback and collaboration from the community before a final decision is made.</dd>

  <dt>Proposal champion:</dt>
  <dd>An individual who takes ownership of advancing a proposal through the RFC process. This role is self-nominated and open to anyone, including both project maintainers and other community members. The champion may be the original author of the proposal or someone who joins later. Responsibilities include drafting the detailed RFC, managing and integrating community feedback, and helping to guide the implementation of the proposal.</dd>
</dl>

:::note

> **You are not alone as a champion!** If this is your first time writing an RFC or design document, our maintainer team will be available to work with you and guide you through this process.

Reviewer 2: ```suggestion
**You are not alone as a champion!** If this is your first time writing an RFC or design document, our maintainers team will guide you through this process.
```

I propose to reduce the scope and add this new responsibility in the `#responsibilities` of `maintainers` in the `governance` page.

Reviewer: Sounds good to me

- nf-core/tools test datasets
  - I would like to propose a new subcommand to the nf-core tools package, called `nf-core testdatasets`. The purpose of this subcommand is to make it easier for developers to search through the very large and convoluted nf-core/test-datasets repository for suitable test-data files to promote reuse of existing data files.
- Standardise `ext.args` naming
  - There is a deficiency within the nf-core/modules specifications regarding standardised use of the `ext.args` variable. The use of the naming scheme `args`, `args2`, `args3` are not intuitive for developers to know which `arg` goes to which command in the module. We propose to update the specifications to use the command/subcommand as a suffix instead e.g., `args_samtoolssort`. This would require a change to every module in the repository and improved documentation support.

:::

As a rule of thumb, if your proposal will require changes across two or more different repositories, it is a good candidate for an RFC.
Projects that are smaller in scope should typically be raised as a new issue on the relevant GitHub repository instead.

# Stage 1: Suggestion (optional)

### Goal

> Provide a space for informal, low-barrier discussions around ideas and potential improvements to nf-core. This stage is useful for gathering early feedback and assessing community interest from the community, maintainers, and other stakeholders.

Reviewer: ```suggestion
Provide a space for informal, low-barrier discussions around ideas and potential improvements to nf-core.
This stage is useful for gathering early feedback and assessing community interest from the community, maintainers, and other stakeholders.
```

One line per sentence as you tell me to do @ewels ;)

Author: Hah, copied as Markdown from Google Docs in this case - can you tell? 😅

### Requirements

> None! Anyone can suggest anything. Proposals should be applicable to the whole community (not pipeline-specific) and large enough to warrant an RFC and associated discussion (not “fix a typo”).

Reviewer: ```suggestion
None! Anyone can suggest anything. Proposals should be applicable to the whole community (not pipeline-specific) and [large enough to warrant an RFC](#scope) and associated discussion (not “fix a typo”).
```

### After acceptance

Once a proposal is accepted, the champion will be invited to write a formal RFC and begin initial development.

A stale, accepted proposal may be rejected retroactively if progress is not made within a reasonable timeframe.

# Stage 3: RFC & Development (the “how”)

### Goal

Collecting (technical) feedback on the implementation of the proposal. Collaborate closely with maintainers and other community stakeholders during development.

### Requirements

A previously accepted proposal from Stage 2 and a confirmed proposal champion to author and implement the RFC.

### Location

RFCs are submitted as GitHub Pull Requests to the nf-core/proposals repository.

> - [See all in-progress RFCs](https://github.com/withastro/roadmap/pulls)

Reviewer 2: good catch
